vshard_shadow_test.go

```diff
@@ -2,28 +2,49 @@ package vshard_router_test
 
 import (
 	"context"
+	"fmt"
 	"testing"
 
 	"github.com/google/uuid"
 	"github.com/stretchr/testify/require"
 
-	vshard_router "github.com/KaymeKaydex/go-vshard-router"
+	vshardrouter "github.com/KaymeKaydex/go-vshard-router"
 	"github.com/KaymeKaydex/go-vshard-router/providers/static"
 )
 
+// Check that ErrorTopologyProvider implements TopologyProvider interface
+var _ vshardrouter.TopologyProvider = (*ErrorTopologyProvider)(nil)
+
+type ErrorTopologyProvider struct{}
+
+func (e *ErrorTopologyProvider) Init(_ vshardrouter.TopologyController) error {
+	return fmt.Errorf("test error")
+}
+func (e *ErrorTopologyProvider) Close() {}
+
+func TestNewRouter_ProviderError(t *testing.T) {
+	ctx := context.TODO()
+	_, err := vshardrouter.NewRouter(ctx, vshardrouter.Config{
+		TotalBucketCount: 256000,
+		TopologyProvider: &ErrorTopologyProvider{},
```

Comment:
We can replace this to and it will work
```golang
TopologyProvider: (*ErrorTopologyProvider)(nil)
```

So we can remove this check above:
```golang
var _ vshardrouter.TopologyProvider = (*ErrorTopologyProvider)(nil)
```

Reply:
done